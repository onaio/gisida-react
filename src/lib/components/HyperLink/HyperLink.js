
import React from 'react';
import PropTypes from 'prop-types';
/**
 * 
 * @param {object} props 
 * To provide more context on the various layer groupings on the menu
 * This component is responsible for rendering links and description icons
 * on the menu
 */
export const HyperLink = (props) => {

  const { link, description, spanClassName, descriptionStyle } = props;
  return (
    link || description ?
    <span className={spanClassName ? spanClassName : "links"}>
      {link ? (
        <a href={link} target="_blank" className="glyphicon glyphicon-list-alt hyperlink"></a>
      ) : null}
      {description ? (
        <div className="description" style={descriptionStyle}>
          <span className="glyphicon glyphicon-info-sign" />
          <p>{description}</p>
        </div>
      ) : (
        ''
      )}
    </span> : null
  );
};
HyperLink.propTypes = {
    link: PropTypes.string,
    description: PropTypes.string,
    spanClassName: PropTypes.string,
    descriptionStyle: PropTypes.string
};
