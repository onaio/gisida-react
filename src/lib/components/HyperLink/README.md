# Hyperlink Component

Providing context to the various sections/subsections on the menu would be important especially for new users. The Hyperlink component compliments the menu sections/subsections by adding links to resources and descriptions to share textual insights on a section/subsections on the menu


# How this works
The component is appended on the menu categories and subcategories on both Menu and Layers/ConnectedLayers respectively. If configs are provided from the site-config the component will return links and description as provided, in the event these aren't provided the component returns null

Usage:
 On the Menu/Layers/ConnectedLayers
```js

<HyperLink
     link={link}
     description={description}
     descriptionStyle={descriptionStyle}
/>
```
