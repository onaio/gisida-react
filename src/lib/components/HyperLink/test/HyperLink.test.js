import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { HyperLink } from '../HyperLink';


describe('components/HyperLink', () => {
        const descriptionStyle =  {
            marginLeft: '45px',
            }
  it('renders without crashing', () => {
    const shallowRender = shallow(
        <HyperLink
            link='https://hello.com'
            description='hello'
            spanClassName='sub-category'
            descriptionStyle={descriptionStyle} 
        />
    );
    expect(toJson(shallowRender)).toMatchSnapshot();
  });

  it('Matches props snapshot', () => {
    const wrapper = mount(
        <HyperLink 
            link='https://hello.com'
            description='hello'
            spanClassName='sub-category' 
            descriptionStyle={descriptionStyle} 
        />
    );
    expect(toJson(wrapper.find('HyperLink'))).toMatchSnapshot();
    wrapper.unmount();
  });
});
