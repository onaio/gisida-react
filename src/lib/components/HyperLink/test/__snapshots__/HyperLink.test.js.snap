// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/HyperLink Matches props snapshot 1`] = `
<HyperLink
  description="hello"
  descriptionStyle={
    Object {
      "marginLeft": "45px",
    }
  }
  link="https://hello.com"
  spanClassName="sub-category"
>
  <span
    className="sub-category"
  >
    <a
      className="glyphicon glyphicon-list-alt hyperlink"
      href="https://hello.com"
      target="_blank"
    />
    <div
      className="description"
      style={
        Object {
          "marginLeft": "45px",
        }
      }
    >
      <span
        className="glyphicon glyphicon-info-sign"
      />
      <p>
        hello
      </p>
    </div>
  </span>
</HyperLink>
`;

exports[`components/HyperLink renders without crashing 1`] = `
<span
  className="sub-category"
>
  <a
    className="glyphicon glyphicon-list-alt hyperlink"
    href="https://hello.com"
    target="_blank"
  />
  <div
    className="description"
    style={
      Object {
        "marginLeft": "45px",
      }
    }
  >
    <span
      className="glyphicon glyphicon-info-sign"
    />
    <p>
      hello
    </p>
  </div>
</span>
`;
